### ISSUE TITLE
[MNG-6590] DefaultProjectArtifactsCache java.lang.IllegalStateException: Duplicate artifact resolution result

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#8661]
**[Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy)** opened **[MNG-6590](https://issues.apache.org/jira/browse/MNG-6590?redirect=false)** and commented

```java
[ERROR] Duplicate artifact resolution result for project org.eclipse.jetty:jetty-project:9.4.15-SNAPSHOT

java.lang.IllegalStateException: Duplicate artifact resolution result for project org.eclipse.jetty:jetty-project:9.4.15-SNAPSHOT

at org.apache.maven.project.artifact.DefaultProjectArtifactsCache.assertUniqueKey (DefaultProjectArtifactsCache.java:218)

at org.apache.maven.project.artifact.DefaultProjectArtifactsCache.put (DefaultProjectArtifactsCache.java:204)

at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.resolveProjectDependencies (LifecycleDependencyResolver.java:149)

at org.apache.maven.lifecycle.internal.MojoExecutor.ensureDependenciesAreResolved (MojoExecutor.java:248)

at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:202)

at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)

at org.apache.maven.lifecycle.internal.[INFO] Downloading from build-master-nexus: http://10.0.0.10:8081/repository/maven-public/org/codehaus/plexus/plexus-utils/3.1.0/plexus-utils-3.1.0.pom

MojoExecutor.execute (MojoExecutor.java:148)

at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)

at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:200)

at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:196)

at java.util.concurrent.FutureTask.run (FutureTask.java:264)

at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:515)

at java.util.concurrent.FutureTask.run (FutureTask.java:264)

at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1128)

at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:628)

at java.lang.Thread.run (Thread.java:834)

[ERROR] java.lang.IllegalStateException: Duplicate artifact resolution result for project org.eclipse.jetty:jetty-project:9.4.15-SNAPSHOT

java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Duplicate artifact resolution result for project org.eclipse.jetty:jetty-project:9.4.15-SNAPSHOT

at java.util.concurrent.FutureTask.report (FutureTask.java:122)

at java.util.concurrent.FutureTask.get (FutureTask.java:191)

at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder.multiThreadedProjectTaskSegmentBuild (MultiThrea[INFO] Downloaded from build-master-nexus: http://10.0.0.10:8081/repository/maven-public/org/codehaus/plexus/plexus-utils/3.1.0/plexus-utils-3.1.0.pom (4.7 kB at 1.2 MB/s)

dedBuilder.java:140)

at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder.build (MultiThreadedBuilder.java:101)

at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)

at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)

at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)

at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)

at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)

at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)

at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)

at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)

at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)

at jdk.[INFO] Downloading from build-master-nexus: http://10.0.0.10:8081/repository/maven-public/org/codehaus/plexus/plexus/4.0/plexus-4.0.pom

internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)

at java.lang.reflect.Method.invoke (Method.java:566)

at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:289)

at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:229)

at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:415)

at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:356)

Caused by: java.lang.IllegalStateException: Duplicate artifact resolution result for project org.eclipse.jetty:jetty-project:9.4.15-SNAPSHOT

at org.apache.maven.project.artifact.DefaultProjectArtifactsCache.assertUniqueKey (DefaultProjectArtifactsCache.java:218)

at org.apache.maven.project.artifact.DefaultProjectArtifactsCache.put (DefaultProjectArtifactsCache.java:204)

at org.apache.maven.lifecycle.internal.LifecycleDependencyResolver.resolveProjectDependencies (LifecycleDependencyResolver.java:149)

at org.apache.maven.lifecycle.internal.MojoExecutor.ensureDependenciesAreResolved (MojoExecutor.java:248)

at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:202)

at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)

at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)

at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)

at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuil[INFO] Downloaded from build-master-nexus: http://10.0.0.10:8081/repository/maven-public/org/codehaus/plexus/plexus/4.0/plexus-4.0.pom (22 kB at 5.4 MB/s)

der$1.call (MultiThreadedBuilder.java:200)

at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call (MultiThreadedBuilder.java:196)

at java.util.concurrent.FutureTask.run (FutureTask.java:264)

at java.util.concurrent.Executors$RunnableAdapter.call (Executors.java:515)

at java.util.concurrent.FutureTask.run (FutureTask.java:264)

at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1128)

at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:628)

at java.lang.Thread.run (Thread.java:834)

[ERROR]

[ERROR] Re-run Maven using the -X switch to enable full debug logging.
```

To reproduce use branch 

jetty-9.4.x_2878_aggregated_javadoc_jar from [https://github.com/eclipse/jetty.project.git](https://github.com/olamy/jetty.project.git)

and build with 

mvn -V -B -T3 -e -Dmaven.test.failure.ignore=true -Djetty.testtracker.log=true install javadoc:javadoc javadoc:aggregate-jar -DskipTests -Paggregate-javadoc-jar

---

**Affects:** 3.6.0

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Since there is no defined fix for that, I'd rather reopen and say cannot reproduce.

**Comment by jira-importer** [comment]
**[Olivier Lamy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=olamy)** commented

"cannot reproduce"  seems wrong as it's very easy to reproduce to 3.6.0 :)

we cannot associate a commit id but at least we have it here for the record (if someone google the stacktrace) as fixed with 3.6.1

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

We could with git bisect 😃
